cli/src/cmd_support_bundle.rs

```diff
@@ -0,0 +1,238 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// Copyright 2025 Oxide Computer Company
+
+use anyhow::bail;
+use anyhow::Context as _;
+use anyhow::Result;
+use async_trait::async_trait;
+use bytes::Buf;
+use bytes::Bytes;
+use camino::Utf8Path;
+use camino::Utf8PathBuf;
+use clap::Parser;
+use futures::Stream;
+use futures::StreamExt;
+use oxide::Client;
+use oxide::ClientHiddenExt;
+use std::io;
+use std::pin::Pin;
+use std::task::Context;
+use std::task::Poll;
+use support_bundle_viewer::BoxedFileAccessor;
+use support_bundle_viewer::LocalFileAccess;
+use support_bundle_viewer::SupportBundleAccessor;
+use support_bundle_viewer::SupportBundleIndex;
+use tokio::io::AsyncRead;
+use tokio::io::AsyncWriteExt;
+use tokio::io::ReadBuf;
+use uuid::Uuid;
+
+/// Downloads a support bundle
+#[derive(Parser, Debug, Clone)]
+#[command(verbatim_doc_comment)]
+#[command(args_conflicts_with_subcommands(true))]
+pub struct CmdDownload {
+    /// ID of the bundle
+    #[clap(long)]
+    id: Uuid,
+
+    /// Path where the bundle should be downloaded
+    #[clap(short, long)]
+    output: Utf8PathBuf,
+}
+
+#[async_trait]
+impl crate::AuthenticatedCmd for CmdDownload {
+    fn is_subtree() -> bool {
+        false
+    }
+
+    async fn run(&self, client: &Client) -> Result<()> {
+        let mut output = tokio::fs::File::create(&self.output)
+            .await
+            .with_context(|| format!("Cannot create output file: {}", self.output))?;
+
+        // NOTE: It might be worth adding a progress bar here?
+        let mut stream = client
+            .support_bundle_download()
+            .support_bundle(self.id)
+            .send()
+            .await?
+            .into_inner_stream();
+
+        while let Some(data) = stream.next().await {
+            match data {
+                Err(err) => bail!(err),
+                Ok(data) => output.write_all(&data).await?,
+            }
+        }
+        Ok(())
+    }
+}
+
+/// Inspects a support bundle
+///
+/// Support bundles may be inspected before they are downloaded (via
+/// smaller HTTP requests), or after the entire zip file has been
+/// downloaded.
+#[derive(Parser, Debug, Clone)]
+#[command(verbatim_doc_comment)]
+#[command(args_conflicts_with_subcommands(true))]
+pub struct CmdInspect {
+    /// ID of the bundle to be inspected, if accessing via API
+    #[clap(long)]
+    id: Option<Uuid>,
+
+    /// Path of the bundle to inspect, if downloaded
+    #[clap(short, long)]
+    path: Option<Utf8PathBuf>,
+}
+
+#[async_trait]
+impl crate::AuthenticatedCmd for CmdInspect {
+    fn is_subtree() -> bool {
+        false
+    }
```

Comment:
I think you can leave both of these as the default (i.e. true) since we don't expect any other subcommands added to them (unless you arrived to this by experimentation--clearly I need to improve the docs on these functions).

Reply:
Sure thing, updated in https://github.com/oxidecomputer/oxide.rs/pull/1093/commits/7704fea4e12af35cb223db87e96f1c31923de8b3